README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# op<b style='color:blue'>uy</b>

*Nicolás Schmidt, Daniela Vairo, Martín Opertti,
UMAD<sup><a id="fnr.1" class="footref" href="#fn.1">1</a></sup>.*

<!-- badges: start -->

[![CRAN
status](https://www.r-pkg.org/badges/version/opuy)](https://CRAN.R-project.org/package=opuy)
[![R build
status](https://github.com/Nicolas-Schmidt/opuy/workflows/R-CMD-check/badge.svg)](https://github.com/Nicolas-Schmidt/opuy/actions)
[![Project Status: Active – The project has reached a stable, usable
state and is being actively
developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
[![](https://img.shields.io/badge/devel%20version-0.1.1001-blue.svg)](https://github.com/Nicolas-Schmidt/opuy)
[![License:
MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
[![DOI](https://zenodo.org/badge/253661638.svg)](https://zenodo.org/badge/latestdoi/253661638)

<!-- badges: end -->

> Datos de opinón pública de Uruguay de 1989 hasta la actualidad.

### Descripción

Provee un conjunto de datos de opinión publica en Uruguay para el
periodo que va desde 1989 hasta la actualidad de dos indicadores
relevante para el estudio de la realidad política del país: **intención
de voto** y **evaluación de gestión del presidente**.

El manual del paquete se puede encontrar
[**aquí**](https://github.com/Nicolas-Schmidt/opuy/blob/master/man/figures/Manual_opuy.pdf).

### Instalación

``` r
## Versión en desarrollo
remotes::install_github("Nicolas-Schmidt/opuy")
```

### Contenido del paquete

#### Funciones

| Nombre         | Descripción                                                             |
|----------------|-------------------------------------------------------------------------|
| `resumen_opuy` | Resumen de los indicadores que contiene la base de datos por consultora |

#### Conjuntos de datos

| Nombre | Descripción                                                                                                                                                                                                                                                                                                                                                                                                  |
|--------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `opuy` | Conjunto de datos que contiene dos indicadores de opinión pública relevante para trabajar con datos políticos para el período 1989 hasta la actualdiad: **Intención de voto** y **Evaluación de la gestión del Presidente**. Los valores para ambos indicadores son agregados ya que no se cuenta con los microdatos. La fuente de los datos es el Banco de Datos de la Faculta de Ciencias Sociales (UMAD). |

#### Ejemplo

``` r
## Conjunto de datos: 'opuy'

library(opuy)

resumen_opuy()
#> 
#> 
#> ---- Resumen de Indicadores por anio y consultora ---------------------------
#> 
#> 
#>            Cifra     Equipos      Factum Interconsult      Opcion       Radar
#> ---- ----------- ----------- -----------  ----------- ----------- -----------
#> 1989       - | -       IV| -       - | -        - | -       - | -       - | -
#> 1990       - | -       IV|EG       - | -        - | -       - | -       - | -
#> 1991       - | -       IV|EG       - | -        - | -       - | -       - | -
#> 1992       - | -       IV|EG       - | -        - | -       - | -       - | -
#> 1993       IV| -       IV|EG       IV| -        - | -       - | -       - | -
#> 1994       IV| -       IV|EG       IV| -        - | -       - | -       - | -
#> 1995       - | -       - |EG       IV| -        - | -       - | -       - | -
#> 1996       - | -       - |EG       IV| -        - | -       - | -       - | -
#> 1997       - | -       IV|EG       IV| -        - | -       - | -       - | -
#> 1998       - | -       IV|EG       IV| -        IV| -       - | -       IV| -
#> 1999       IV| -       IV|EG       IV| -        IV| -       - | -       IV| -
#> 2000       - |EG       - |EG       IV|EG        - | -       - | -       - | -
#> 2001       - | -       - |EG       IV|EG        - | -       - | -       - | -
#> 2002       - | -       IV|EG       IV|EG        - | -       - | -       - | -
#> 2003       IV| -       - |EG       IV| -        IV| -       - | -       IV| -
#> 2004       IV| -       IV|EG       IV| -        IV| -       - | -       IV| -
#> 2005       - |EG       - |EG       - | -        IV|EG       - | -       - |EG
#> 2006       - | -       - |EG       - |EG        IV|EG       - | -       - | -
#> 2007       - | -       - |EG       IV| -        IV|EG       - | -       - | -
#> 2008       IV|EG       IV|EG       IV| -        IV|EG       - | -       IV| -
#> 2009       IV| -       IV|EG       IV| -        IV| -       - | -       IV| -
#> 2010       - |EG       IV|EG       IV|EG        - |EG       - | -       - | -
#> 2011       IV|EG       IV|EG       IV| -        - | -       - | -       - | -
#> 2012       IV|EG       IV|EG       IV| -        - | -       - | -       IV|EG
#> 2013       IV|EG       IV|EG       IV| -        IV| -       - | -       IV| -
#> 2014       IV|EG       IV|EG       IV| -        IV| -       IV|EG       IV| -
#> 2015       - |EG       IV|EG       - |EG        - | -       - |EG       - | -
#> 2016       IV|EG       IV|EG       IV|EG        IV| -       IV|EG       - | -
#> 2017       IV|EG       IV|EG       IV|EG        IV| -       IV|EG       IV| -
#> 2018       IV|EG       IV|EG       IV| -        - | -       IV|EG       IV| -
#> 2019       IV|EG       IV|EG       IV| -        - | -       IV|EG       IV| -
#> 2020       - |EG       - |EG       - | -        - | -       - |EG       - | -
#> 2021       - |EG       - |EG       - |EG        - | -       - |EG       - | -
#> 2022       - |EG       - |EG       - | -        - | -       - |EG       - | -
#> 
#> IV: Intencion de Voto
#> EG: Evaluacion de gestion
#> 
#> -----------------------------------------------------------------------------
```

###### Intención de voto para elecciones nacionales medidas en 2019 por todas las consultoras

``` r
library(tidyverse)

data(opuy)

opuy %>%
    filter(medicion == 'Intencion de voto',
           tipo_eleccion == 'Nacional',
           anio_medicion == 2019, 
           sigla %in% c('FA', 'PC', 'PN', 'CA')) %>% 
    mutate(partido = factor(partido, levels = c('Frente Amplio', 'Partido Colorado', 
                                                'Partido Nacional', 'Cabildo Abierto'))) %>% 
    ggplot(aes(x = fecha, y = valor, color = empresa)) +
    geom_line(aes(group = empresa), size = 1, alpha = 0.6) +
    geom_point(size = 1.5) +
    facet_wrap(~partido, nrow = 1) +
    hrbrthemes::theme_ipsum_tw(grid = "XY", axis = "xy") +
    labs(x = "",
         y = "Porcentaje de votos %",
         color = "",
         title = "Intención de voto en elecciones nacionales 2019",
         subtitle = '27 de octubre de 2019 \nMediciones del año 2019',
         caption = 'Fuente: Unidad de Métodos y Acceso a Datos (UMAD)')
```

<img src="man/figures/README-unnamed-chunk-4-1.png" width="100%" />

###### Intención de voto para elecciones nacionales de todas las consultoras para todo el período de medición de la elección de 2019 (2016 - 2019)

<img src='man/figures/iv2019.gif'/>

##### Resumen intención de voto 2019 de los principales partidos para toda la serie correspondiente a la elección nacional de 2019 (2016 - 2019)

| Partido          | Valor Mínimo | Valor Máximo | Valor Mínimo 2019 | Valor Máximo 2019 | Votación real |
|:-----------------|:-------------|:-------------|:------------------|:------------------|:--------------|
| Frente Amplio    | 24%          | 43%          | 27%               | 43%               | 39.25%        |
| Partido Nacional | 21.6%        | 34%          | 21.6%             | 33%               | 28.79%        |
| Partido Colorado | 3%           | 22%          | 9%                | 22%               | 12.41%        |
| Cabildo Abierto  | 1%           | 12%          | 1%                | 12%               | NA%           |

#### Evaluación de la gestión del presidente

``` r
opuy %>%
    filter(medicion == 'Evaluacion de gestion presidente',
           categoria_unificada == 3) %>%
    select(anio_gobierno, empresa, valor, presidente) %>%
    group_by(empresa, anio_gobierno, presidente) %>%
    summarise(promedio = mean(valor, na.rm = TRUE)) %>%
    ungroup() %>%
    mutate(presidente = factor(presidente, levels = c("Lacalle", "Sanguinetti 2", 
                                                      "Batlle", "Vazquez 1", "Mujica", 
                                                      "Vazquez 2", "Lacalle Pou"))) %>% 
    ggplot(aes(x = factor(anio_gobierno), y = promedio, color = empresa)) +
    geom_line(aes(group = empresa), size = 1, alpha = 0.6) +
    geom_point(size = 1.5) +
    facet_wrap(~presidente, nrow = 1) +
    hrbrthemes::theme_ipsum_tw(grid = "XY", axis = "xy") +
    labs(x = "Año de gobierno",
         y = "Porcentaje de aprobación",
         color = "", 
         title = "Evaluacion de la gestión del presidente",
         subtitle = 'Serie histórica con datos de todas las consultoras (promedios anuales)',
         caption = 'Fuente: Unidad de Métodos y Acceso a Datos (UMAD)')
```

<img src="man/figures/README-unnamed-chunk-6-1.png" width="100%" />

#### Citar

Para citar el paquete `opuy` o el uso de los datos en publicaciones use:

**Schmidt, Nicolás, Vairo, Daniela, Opertti, Martín , UMAD (2022) opuy:
Datos de Opinión Pública de Uruguay 1989 a la actualidad, R package
version 0.1.1001, <https://nicolas-schmidt.github.io/opuy/>.**

``` r
citation(package = 'opuy')
#> 
#> 
#> A BibTeX entry for LaTeX users is
#> 
#>   @Manual{,
#>     title = {opuy: Datos de Opinion Publica de Uruguay desde 1989},
#>     author = {Nicolas Schmidt and Daniela Vairo and Martin Opertti and {UMAD}},
#>     year = {2020},
#>     note = {R package version 0.1.1001},
#>     url = {https://nicolas-schmidt.github.io/opuy/},
#>   }
```

------------------------------------------------------------------------

##### Mantenedor

Nicolás Schmidt (<nschmidt@cienciassociales.edu.uy>)

#### Notas

------------------------------------------------------------------------

<sup><a id="fn.1" href="#fnr.1">1</a></sup> Unidad de Métodos y Acceso a
Datos, Facultad de Ciencias Sociales, Universidad de la República
(UMAD-FCS-UdelaR)


NEWS.md
# opuy 0.1.1001

* Se agregan datos hasta agosto de 2022

# opuy 0.1.1

* Lanzamiento del paquete con datos desde 1989 hasta 2020


R/opuy.R
#'opuy
#'
#' Contiene dos indicadores relevantes de opinión publica relativos a las
#' elecciones y al gobierno en Uruguay desde 1989 hasta la actualdiad: intención de voto y evaluación
#' de la gestión del presidente. Los datos son agregados y provienen de la divulgación de los mismos por parte
#' de las empresas consultoras de opinion publica que realizan las encuestas.
#'
#'  Los datos fueron obtenidos por UMAD en base a comunicaciones publicas
#'  (medios de prensa fundamentalmente) de las distintas consultoras. La estructura, orden, armonizacion y
#'  creacion de nuevas variables es desarrollo de los autores del paquete.
#'
#'\describe{
#'    \item{medicion}{Indicador disponible. Hasta el momento solo hay dos: \emph{Intención de voto} y \emph{Evaluación de la gestión del Presidente}.}
#'    \item{empresa}{Nombre de la empresa consultora que realizó la medición.}
#'    \item{tipo_eleccion}{Tipo de elección: \emph{Nacional}, \emph{Balotaje} o \emph{Internas}. Esta variable es útil para el indicador 'Intención de voto'.}
#'    \item{anio_medicion}{Anio en el que se publicó la medición.}
#'    \item{anio_gobierno}{Anio de gobierno. Esta variable es un número entero que va desde el 1 hasta el 5 indicando respectivamente el anio de gobierno (1 = primer anio de gobierno, 2 = segundo anio de gobierno...). Es particularmente útil para el indicador 'Evaluación de la gestión del Presidente'. Con esta variable es más simple realizar comparaciones entre Presidentes según anio de gobierno.}
#'    \item{eleccion}{Esta variable refiere al anio de la elección a la que está asociada la medición. Por ejemplo, se puede tener una medición de Intención de voto del anio 2017, y esta corresponde a la eleccion de 2019.}
#'    \item{fecha}{Fecha de cierre de campo declarado por la empresa consultora. En algunos casos es posible que la fecha sea la de publicación de la medición.}
#'    \item{partido}{Nombre del partido político.}
#'    \item{sigla}{Sigla del partido político.}
#'    \item{candidato}{Nombre del candidato para el caso de la Intención de Voto en elecciones Internas y en Balotaje.}
#'    \item{presidente}{Nombre del Presidente para la medición Evaluación de la gestión del Presidente.}
#'    \item{categoria}{Categoría original de la medición para el caso de Evaluación de la gestión del Presidente. Esta variable comúnmente se mide en escala de Likert (e.g. Muy Mala, Mala, Regular, Buena, Muy buena)}
#'    \item{categoria_unificada}{Se armoniza la variable 'categoria' para poder comparar entre empresas consultoras. Los encuestados que no respondieron o que respondieron que no saben se les dió el valor de 0, los que desaprueban asumen el valor de 1, los que son indiferentes el valor de 2 y con el valor de 3 son los encuestados que respondieron que aprobaban la gestión del Presidente.}
#'    \item{valor}{Valor del indicador en porcentaje.}
#' }
#' @docType data
#' @keywords datasets
#' @name opuy
#' @usage data(opuy)
#' @format data.frame con 4956 filas y 14 columnas
'opuy'


R/resumen_opuy.R

#' @title resumen opuy
#' @description Resumen de los indicadores que contiene la base de datos por consultora.
#' @return data.frame que contiene los indicadores que cada consultora realizó por anio.
#' @examples
#' resumen_opuy()
#' @export


resumen_opuy <- function(){

    op <- opuy %>%
        select(anio_medicion, empresa, medicion) %>%
        distinct() %>%
        group_by(anio_medicion, empresa) %>%
        mutate(
            medicion  = ifelse(medicion == 'Intencion de voto', '  IV| -', '  - |EG'),
            indicador = ifelse(length(medicion) == 2, "  IV|EG", medicion)) %>%
        ungroup() %>%
        select(-medicion) %>%
        distinct() %>%
        pivot_wider(., names_from = empresa, values_from = indicador) %>%
        arrange(anio_medicion) %>%
        as.data.frame()

    rownames(op) <- op$anio_medicion
    op <- op[, -1]
    op[is.na(op)] <- "  - | -"
    op <- rbind(paste(rep("-", 11), collapse = ""), op)
    rownames(op)[1] <- "----"

    cat("\n\n---- Resumen de Indicadores por anio y consultora ---------------------------\n\n\n")
    print(op)
    cat("\nIV: Intencion de Voto\nEG: Evaluacion de gestion\n\n")
    cat("-----------------------------------------------------------------------------\n")
}




data-raw/datos_op_uy.R

## -----------------------------------------------------------------------------
## op_uy
## -----------------------------------------------------------------------------

#opuy <- readxl::read_excel("data-raw/opuy.xlsx")

opuy <- readxl::read_excel("data-raw/opuy.xlsx",
                           col_types = c("text", "text", "text", "numeric",
                                         "numeric", "numeric", "numeric",
                                         "text", "text", "text", "text",
                                         "text", "numeric", "numeric"))

str(opuy)
opuy$fecha <- janitor::excel_numeric_to_date(opuy$fecha)
opuy <- dplyr::arrange(opuy, medicion, empresa, fecha)
rio::export(opuy, 'opuy.rda')
load('opuy.rda')
opuy <- tibble::as_tibble(opuy)
save(opuy, file = here::here("data", "opuy.rda"))
file.remove('opuy.rda')


data-raw/op.R

library(gganimate)
library(tidyverse)
library(readxl)
library(Boreluy)
library(opuy)
library(hrbrthemes)
library(extrafont)

#font_import()
#loadfonts(device = "win")

una <- opuy %>%
    filter(medicion == 'Intencion de voto',
             tipo_eleccion == 'Nacional',
             eleccion == 2019,
           sigla %in% c('FA', 'PC', 'PN', 'CA')) %>%
    arrange(empresa, fecha)


una$partido <- factor(una$partido)
una$partido <- factor(una$partido, levels = levels(una$partido)[c(2, 3, 4, 1)])

una <- una %>%
    group_by(partido) %>%
    mutate(minimo = min(valor, na.rm = TRUE),
           maximo = max(valor, na.rm = TRUE)) %>%
    ungroup()

franja <- distinct(una[, c('partido', 'minimo', 'maximo')])


g2 <-
    ggplot(una, aes(x = fecha, y = valor, color = empresa)) +
    geom_line(aes(group = empresa), size = 1, alpha = 0.6) +
    geom_point(size = 1.5) +
    geom_segment(aes(xend = fecha[length(fecha)], yend = valor, group = empresa),
                 linetype = 2, colour = 'grey') +
    geom_text(aes(x = fecha[length(fecha)], label = empresa), hjust = 0) +
    ylim(0, 60) +
    #scale_color_ipsum() +
    #scale_fill_ipsum() +
    facet_wrap(~partido, nrow = 1) + #, scales = "free") +
    geom_hline(aes(yintercept = minimo), franja, color = 'red', alpha = 1, linetype = "dotted") +
    geom_hline(aes(yintercept = maximo), franja, color = 'red', alpha = 1, linetype = "dotted") +
    guides(color = FALSE) +
    theme_ipsum_tw(grid="XY", axis="xy") +
    labs(x = "",
         y = "Porcentaje de votos",
         title = "Intención de voto en elecciones nacionales 2019",
         subtitle = '27 de octubre de 2019',
         caption = 'Fuente: Unidad de Métodos y Acceso a Datos (UMAD)') +
    theme(plot.title = element_text(size = 15),
          plot.subtitle = element_text(size = 12),
          axis.text.x = element_text(size = 10),
          plot.caption = element_text(size = 10, color = "grey40")) +
    transition_reveal(fecha) +
    coord_cartesian(clip = 'off') +
    enter_fade() +
    exit_shrink()

#animate(g2, duration = 25)
animate(g2, height = 500, width = 800, duration = 20)
anim_save("C:\\Users\\usuario\\Desktop\\R_packages\\opuy\\opuy\\man\\figures\\iv2019.gif")



# ADD text
e19  <- resultado_eleccion_uy(2019, 'Presidencial') %>%
    select(Partido, Porcentaje) %>%
    rename(partido = Partido)

una2 <- una %>%
    group_by(partido) %>%
    summarise(minimo = min(valor),
              maximo = max(valor)) %>%
    ungroup() %>%
    mutate(partido = as.character(partido)) %>%
    left_join(., e19, by = 'partido') %>%
    mutate(
        text = paste('Resultado...', Porcentaje)
    )
una2$partido <- factor(una2$partido)
una2$partido <- factor(una2$partido, levels = levels(una2$partido)[c(2, 3, 4, 1)])



## tabla
e19  <- resultado_eleccion_uy(2019, 'Presidencial') %>%
    select(Partido, Porcentaje)
tab <- opuy %>%
    filter(medicion == 'Intencion de voto',
           tipo_eleccion == 'Nacional',
           eleccion == 2019,
           sigla %in% c('FA', 'PC', 'PN', 'CA')) %>%
    group_by(partido) %>%
    summarise('Valor Mínimo' = min(valor),
              'Valor Máximo' = max(valor),
              'Valor Mínimo 2019' = min(valor[anio_medicion == 2019]),
              'Valor Máximo 2019' = max(valor[anio_medicion == 2019])) %>%
    ungroup() %>%
    rename(Partido = partido) %>%
    left_join(., e19, by = 'Partido') %>%
    arrange(-Porcentaje) %>%
    rename('Votacón real' = Porcentaje)

tab <- cbind(tab[,1], apply(tab[, -1], 2, function(x){paste0(x, "%")}))
tab



## agrupar categoria
library(opuy)
unique(opuy$categoria)


library(tidyverse)
data(opuy)


print(
opuy %>%
    filter(medicion == 'Evaluacion de gestion presidente',
           categoria_unificada == 3) %>%
    select(anio_gobierno, empresa, valor, presidente) %>%
    group_by(empresa, anio_gobierno, presidente) %>%
    summarise(promedio = mean(valor, na.rm = TRUE)) %>%
    ungroup() %>%
    na.omit() %>%
    arrange(presidente) %>%
    mutate(anio_gobierno = paste("Año", anio_gobierno))

    , n = 100)


opuy %>%
    filter(medicion == 'Evaluacion de gestion presidente',
           categoria_unificada == 3) %>%
    select(anio_medicion, empresa, valor, presidente) %>%
    group_by(empresa, anio_medicion, presidente) %>%
    summarise(promedio = mean(valor, na.rm = TRUE)) %>%
    ungroup() %>%
    na.omit() %>%
    arrange(presidente) %>%
    mutate(presidente = factor(presidente, levels = c("Lacalle", "Sanguinetti 2",
                                                      "Batlle", "Vazquez 1", "Mujica",
                                                      "Vazquez 2", "Lacalle Pou"))) %>%
    ggplot(aes(x = anio_medicion, y = promedio, color = empresa)) +
    geom_line(aes(group = empresa), size = 1, alpha = 0.6) +
    geom_point(size = 1.5) +
    facet_wrap(~presidente, nrow = 1) +
    theme_ipsum_tw(grid="XY", axis="xy") +
    labs(x = "Año de gobierno",
         y = "Porcentaje de aprobación",
         color = "",
         title = "Evaluacion de la gestión del presidente",
         subtitle = 'Serie historica con datos de todas las consultoras',
         caption = 'Fuente: Unidad de Métodos y Acceso a Datos (UMAD)')
